File: src/C#/DotNETTWAINDemo/DotNETTWAINDemo/ZoomForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DotNet_TWAIN_Demo
{
    public partial class ZoomForm : Form
    {
        private int zoomRatio;
        
        public ZoomForm() : this(1)
        {
            
        }

        public ZoomForm(float ratio)
        {
            InitializeComponent();
            ZoomRatio = ratio;
            this.DialogResult = DialogResult.Cancel;

        }

        public float ZoomRatio
        {
            get { return zoomRatio / 100F; }
            set 
            {
                zoomRatio = (int)(value * 100 + 0.1);
                this.tbxZoomRatio.Text = zoomRatio.ToString();
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            int ratio = -1;

            if (!int.TryParse(tbxZoomRatio.Text, out ratio) || ratio < 2 || ratio > 6500)
            {
                tbxZoomRatio.Focus();
                return;
            }
            else
                zoomRatio = ratio;

            this.DialogResult = DialogResult.OK;
            this.Close();
        }

        private void tbxZoomRatio_KeyPress(object sender, KeyPressEventArgs e)
        {
            if ((int)e.KeyChar == 13)
                btnOK_Click(this, null);
        }
    }
}

File: samples/VB .NET Samples/DotNETTWAINDemo/DotNETTWAINDemo/ZoomForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class ZoomForm
	Inherits Form
	Private m_zoomRatio As Integer

	Public Sub New()

		Me.New(1)
	End Sub

	Public Sub New(ratio As Single)
		InitializeComponent()
		ZoomRatio = ratio

		Me.DialogResult = DialogResult.Cancel
	End Sub

	Public Property ZoomRatio() As Single
		Get
			Return m_zoomRatio / 100F
		End Get
		Set
			m_zoomRatio = CInt(Math.Truncate(value * 100 + 0.1))
			Me.tbxZoomRatio.Text = m_zoomRatio.ToString()
		End Set
	End Property

	Private Sub btnCancel_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub

	Private Sub btnOK_Click(sender As Object, e As EventArgs)
		Dim ratio As Integer = -1

		If Not Integer.TryParse(tbxZoomRatio.Text, ratio) OrElse ratio < 2 OrElse ratio > 6500 Then
			tbxZoomRatio.Focus()
			Return
		Else
			m_zoomRatio = ratio
		End If

		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub

	Private Sub tbxZoomRatio_KeyPress(sender As Object, e As KeyPressEventArgs)
		If AscW(e.KeyChar) = 13 Then
			btnOK_Click(Me, Nothing)
		End If
	End Sub
End Class
